libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/TriggerConditions/OnCondition.cs

```diff
@@ -146,6 +225,12 @@ public void AddExternalCondition(string condition)
         /// <returns>A <see cref="Task"/> with plan change list.</returns>
         public virtual async Task<List<ActionChangeList>> ExecuteAsync(SequenceContext planningContext)
         {
+            if (RunOnce)
+            {
+                var count = planningContext.GetState().GetValue<uint>(DialogPath.EventCounter);
+                planningContext.GetState().SetValue(AdaptiveDialog.ConditionTracker + "." + Id.ToString() + ".lastRun", count);
```

Comment:
>AdaptiveDialog.ConditionTracker + "." + Id.ToString() + ".lastRun" [](start = 52, length = 66)

Nit: String interpolation?  $"{AdaptiveDialog.ConditionTracker}.{Id}.lastrun" is much easier to read